convex/openai.ts

```diff
@@ -0,0 +1,69 @@
+"use node";
+
+import { action } from "./_generated/server";
+import { v } from "convex/values";
+import { api } from "../convex/_generated/api";
+import OpenAI from "openai";
+
+// Initialize the OpenAI client with the given API key
+const openai = new OpenAI({
+  apiKey: process.env.OPENAI_API_KEY, // This is also the default, can be omitted
+});
+// start a game
+export const startGames = action({
+  args: {
+    topic: v.string(),
+    numberOfPlayer: v.number(),
+    roomId: v.id("rooms"),
+  },
+  handler: async (ctx, args) => {
+    try {
+      // ! TODO: check if this acutally works!!!
+      const checkTopicResponse = await openai.completions.create({
+        model: "text-davinci-003",
+        prompt: `if the  topic  '${args.topic}' is a good topic`,
```

Comment:
You have to send prompts as you're talking with other person. For example here it can be:
```javascript
const prompt = `Is the topic '${args.topic}' a good topic?
A good topic must be a single or two word topic like 'animal',
'plants', 'cars', 'celebrities', etc. Anything that can be suitable
for all audiences. Your answert must be just false or true, with
any other word included.`
```

Reply:
This api I can't test, because I can't use it In China,So if you have the pay key, you can fix this problem